### PR TITLE
add_additionalLabels_to_chart

### PR DESCRIPTION
**Pull request by TheCubicleJockey**
# Description
This new feature is to add additionalLabels to helm chart

# Context
See issue [here](https://github.com/solo-io/gloo/issues/9035)

## Testing steps
I manually verified behavior by templating the helm chart with different values

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: install/helm/gloo/templates/_helpers.tpl [hunk 1]
```diff
@@ -263,3 +263,14 @@ Otherwise it will generate ["Create", "Update", "Delete"]
 {{- end -}}
 {{ toJson $result }}
 {{- end -}}
+
+{{/*
+Common labels
+*/}}
+{{- define "gloo.labels" -}}
+gloo: gloo
```
Comment (davidjumani) [inline install/helm/gloo/templates/_helpers.tpl:271]: The `gloo` label indicates components such as proxy, rbac, etc. This should not be part of the default labels

File: install/helm/gloo/templates/0-namespace.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ kind: Namespace
 metadata:
   name: {{ .Release.Namespace }}
   labels:
-    app: gloo
+    {{- include "gloo.labels" (dict "context" .) | nindent 4 }}
```
Comment (davidjumani) [inline install/helm/gloo/templates/0-namespace.yaml:7]: We can pass it as `{{- include "gloo.labels" . }}` and modify `gloo.labels` accordingly

File: install/helm/gloo/templates/_helpers.tpl [hunk 3]
```diff
@@ -263,3 +263,14 @@ Otherwise it will generate ["Create", "Update", "Delete"]
...
+app: gloo
+{{- with .context.Values.additionalLabels }}
```
Comment (davidjumani) [inline install/helm/gloo/templates/_helpers.tpl:273]: Can simplify this to {{- with .Values.additionalLabels }}

**Review by davidjumani — CHANGES_REQUESTED** [review]
The overall idea is good and simplifies the charts, just a few changes are required
- Remove the `gloo: gloo` from the default label set
- For simplicity do not pass the context as a `context` variable but just `.`
- Run the tests in the install directory and try to fix them

File: install/helm/gloo/templates/0-namespace.yaml (hunk 2 above)
Comment (TheCubicleJockey) [reply install/helm/gloo/templates/0-namespace.yaml]: Resolved in next push

File: install/helm/gloo/templates/1-gloo-deployment.yaml [hunk 4]
```diff
@@ -8,8 +8,7 @@ apiVersion: apps/v1
 kind: Deployment
 metadata:
   labels:
-    app: gloo
-    gloo: gloo
+    {{- include "gloo.labels" . }}
```
Comment (davidjumani) [inline install/helm/gloo/templates/1-gloo-deployment.yaml:11]:
```suggestion
    gloo: gloo
    {{- include "gloo.labels" . }}
```
And so for all other occurrences of `gloo: gloo` that has been removed

**Review by davidjumani — CHANGES_REQUESTED** [review]
Thanks for addressing the comments. One last thing - Can you please make the following changes ?

**Comment by davidjumani** [comment]
@TheCubicleJockey Can you please rebase and fix the conflicts as well as revert all occurrences of `gloo: gloo` that have been removed

**Comment by davidjumani** [comment]
Moving this to draft until the enterprise changes are addressed

**Comment by TheCubicleJockey** [comment]
Please let me know what needs done to get this moved over the line. My client (your enterprise client) is wanting this merged.

**Comment by davidjumani** [comment]
Hi @TheCubicleJockey I'm continuing this in a [new PR](https://github.com/solo-io/gloo/pull/9491) for faster development

**Pull request closed without merging**